src/chart/model/scale.model.ts

```diff
@@ -222,6 +232,12 @@ export class ScaleModel extends ViewportModel {
 		const zoomX = this.calculateZoomX(xStart, xEnd);
 		const state = { ...initialState, zoomX, xStart, xEnd };
 		const constrainedState = this.scalePostProcessor(initialState, state);
+		const zoomIn = constrainedState.xStart > initialState.xStart || constrainedState.xEnd < initialState.xEnd;
```

Comment:
`xStart `for constrained state can be moved to left, but viewport can be the same, if  `xEnd `moved to left too.
You should check here `constrainedState.xEnd - constrainedState.xStart< initialState.xEnd - initialState.xStart;`